### DISCUSSION TITLE
media-library:clean runs out of memory on large collection

### DISCUSSION THREAD

**Discussion by marten-wirelab** [spatie/laravel-medialibrary#3332] [Ideas]
Hi,

I have a pretty large media library of about 575k images linked to about 22k distinct objects of one model type, which I'm trying to clean. However, when running the `media-library:clean` command, I get the following error: 
```
PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in [...]/vendor/laravel/framework/src/Illuminate/Database/Connection.php on line 421
```

I think there should be some sort of logic added to limit memory usage.

**Comment by systemsolutionweb** [comment]
Hi, you could do a PR using chunk 

https://laravel.com/docs/10.x/queries#chunking-results
